aws/data_source_aws_iam_assumed_role_source.go

```diff
@@ -0,0 +1,140 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/iam/finder"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/iam/waiter"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/tfresource"
+)
+
+func dataSourceAwsIAMAssumedRoleSource() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsIAMAssumedRoleSourceRead,
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"role_path": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"role_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"source_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"session_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsIAMAssumedRoleSourceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iamconn
+
+	arn := d.Get("arn").(string)
+
+	d.SetId(arn)
+
+	roleName := ""
+	sessionName := ""
+	var err error
+
+	if roleName, sessionName = roleNameSessionFromARN(arn); roleName == "" {
+		d.Set("source_arn", arn)
+		d.Set("session_name", "")
+		d.Set("role_name", "")
+		d.Set("role_path", "")
```

Comment:
I don't think returning `role_path` is necessary. Seems like an arbitrary field to choose to return from GetRole output. Maybe we could leave it off and if folks want more role context for this datasource we can always add it later?

Reply:
Removed.